### ISSUE TITLE
[New rule] validate settings

### ISSUE THREAD

**Issue by dkrutskikh** [dart-code-checker/dart-code-metrics#151]
**Please describe what the rule should do:**

Validate settings block in analysis_options.yaml

**If your rule is inspired by other please provide link to it:**

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Other (please specify:) Warns about problems in plugin setting block

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

```dart

```

**Are you willing to submit a pull request to implement this rule?**
